### PR TITLE
Add Map of RSEs and RSE Groups

### PR DESCRIPTION
**Pull request by vsoch**
This pull request will add a map for RSEs, and RSE groups. If an individual RSE has added his or her entry to the authors.yml, this can be supplemented with a coordinate. If a group or individual (not wanted to be added to the feed) wants to be added to the map, this can be done with the map.yml file. I've created a layer for each on the map, and also added a test file for the data itself. The map looks like this:

image

Notice that you can click to select layers in the upper right (both displayed by default). I've also redone the navigation to have dropdowns, since we now have more than a few links. The "RSE Map" is under the Community tab:

image

And the archive is moved to be under posts.

### CODE REVIEW

File: _includes/map.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+marker.bindPopup("Research Software Engineer<br>{% if author.image %}<img src='{{ author.image }}' style='height:30px; float:right'>{% endif %}<a href='{{ author.url }}'>{{ author.name }}{% if author.institution %}<br>{{ author.institution }}{% endif %}</a>");
+RSEs.push(marker);{% endif %}{% endfor %}
+
+// Add them to a layer group
+var rseLayer = L.layerGroup(RSEs);
+
+// Next add groups, and RSEs without feeds
+{% for entry in site.data.map %}{% if entry.coords %}
+var marker = L.marker([{{ entry.coords[0] }}, {{ entry.coords[1] }}]);
+marker.bindPopup("RSE Group<br>{% if entry.image %}<img src='{{ entry.image }}' style='height:30px; float:right'>{% endif %}<a href='{{ entry.url }}'>{{ entry.name }}{% if entry.institution %}<br>{{ entry.institution }}{% endif %}</a>");
```
Comment (cferenba) [inline _includes/map.html]: If I'm reading this correctly, you've hardcoded in the string `RSE Group`, even in the case of an individual RSE without a feed.  Can we add a way to distinguish the two?

**Comment by cferenba** [comment]
@vsoch : This looks great, thanks for setting it up!

One possible use case that I can't tell whether you've covered or not:  the individual RSE who wants a pin for their location and institution, but doesn't want any personal info posted.  Can we support that?

**Comment by vsoch** [comment]
I don't see why not, you can add either to authors.yml (with feed) or map.yml (without feed), the latter for an individual or institution. What I think you are hinting as is that (currently) all that are added via the map.yml are considered "groups" and this might not be the case. I'll add a variable to indicate RSE or group, give me a few minutes!

**Comment by vsoch** [comment]
okay, here is the new preview! https://28-186452639-gh.circle-artifacts.com/0/blog/map/index.html if you want to see the new tests for the map data, see here https://circleci.com/gh/USRSE/blog/28#tests/containers/0 and the section titled "test the metadata required for map"

image

**Comment by vsoch** [comment]
So now it would be supported to add yourself as an individual under `map.yml`, specify the type to be "person" but still only provide the name / url for your institution.

**Comment by cosden** [comment]
@vsoch great job!  It looks fantastic! As a first iteration, it's terrific.  I think this should live on the main us-rse.org page.  

One comment:  The default heading on the pins is "RSE Groups." I'd take that out so that individuals can put their info in map.yml

**Comment by vsoch** [comment]
Oh yes, good point! I'll have that be based on the type as well. One more second :)

File: _includes/map.html (hunk 1 above)
Comment (cosden) [reply _includes/map.html]: I just said the same thing!  Seems we're on the same page. :)

**Comment by vsoch** [comment]
To be clear, the first line is a title to distinguish the two layers (RSE Groups and RSEs). It's important so that when both are displayed on the map, it's clear if the marker is for a person or group. If the individual wants to add some custom title, they can add it to "name" with a break <br> and pretty much render whatever they please :)

**Comment by cferenba** [comment]
Looks good!

**Comment by cosden** [comment]
I vote we merge it, show it to the group in slack and have people edit/add their entries here.  Then in a few days/week migrate it to the main page.  It's arguably better than the list of groups.

**Comment by vsoch** [comment]
okay, I'll take 2 good reviews as indicator of ready to merge! And I completely agree @cosden  - I've found that getting feedback for ideas is almost impossible, but creating a start of a thing, showing it to people and then (oh no something is wrong/missing on the internet!) is a good incentivizer for contribution. 

Thanks to you both! Merging.

**Comment by cosden** [comment]
@vsoch +1!  That's absolutely the way to get things done.

**Comment by vsoch** [comment]
@cosden I would suggest that the main page, instead of being served at usrseweb, should be at usrse.github.io. That way, the CNAME for the main site will be served (still) from there, but then the "blog" will also (appear to be) served on the same domain. I advocate for leaving the map with the blog, because we are deriving data both from the authors.yml and map.yml. The "main" site is more group / about RSE focused, while the "blog" site is more "here is an RSE community, with actual individuals." As an individual I would be incentivized to add my feed (and coordinate) to this site, whereas I wouldn't be for just a map on the other site. If you are worried about redundant information, then I would suggest serving the sites under the same domain, and having the "groups" page render (from the same data file) but to usrse.github.io/groups. The goal would be to have seamlessness between the two, if that makes sense, and only maintaining one data file.

**Comment by cosden** [comment]
Hi @vsoch 

>  I would suggest that the main page, instead of being served at usrseweb, should be at usrse.github.io.

I like this.  Makes sense.  All that we need to do is rename the repo, right?  Are there there any other ramifications I'm not thinking of?  

> I advocate for leaving the map with the blog, because we are deriving data both from the authors.yml and map.yml. 

I guess I don't really have a strong feeling.  I think I view the syndicated blog site as more a collection of resources, rather than generating or growing a community.  I don't have a personal blog, and while it's unlikely I ever will (let's face it I have very little interesting to say) my group may have a team blog.  I suspect those that have active blogs are in the small minority.  But, the reason I'm really excited about the inclusion of the collection of blogs is that it's a place people can go read and learn about RSE related things.  I view the "main" page as a portal to the community.  Hence, my gut was to show the size of the community directly from the main page.  In other words, "join this group" because there are lots of people like you (some write blogs, many don't). 

How about we leave it with the blog, get the branding/story consistent and then reevaluate in a couple months?

**Comment by vsoch** [comment]
> I like this. Makes sense. All that we need to do is rename the repo, right? Are there there any other ramifications I'm not thinking of?

Off the top of my head, I don't think so. The main ramification would be links. However I'd like to insist that this is not done until there is https. As long as there isn't https, using the CNAME feels a bit sketchy.

> How about we leave it with the blog, get the branding/story consistent and then reevaluate in a couple months?

I agree, and I have a proposal. I think we should create a third repo (a copy of the current main site) and start to slowly fix it up. The main issue is that everything is very manual - adding / updating jobs, writing pages, and nothing is tested. The site should need minimal maintenance, ideally, other than someone adding a data point somewhere via PR when necessary (e.g. a job). It should have regular testing of said data points. If we create this copy then it gives us time to fix it up without worrying that it's public. What do you think?

**Comment by cosden** [comment]
> I think we should create a third repo (a copy of the current main site) and start to slowly fix it up.

So basically split it into dev and prod sites?  And once we get the dev version to where we like it, it gets merged into the prod one?  It wouldn't just replace because there might have been small updates we need to capture, right?  Fantastic idea, and I'm 100% in support of this, as long as the actual implementation is clear and straightforward for contributors.

**Comment by vsoch** [comment]
The issue with having a fork is that we couldn't fork to the same organization. We could, however, duplicate. I agree it's an issue with small updates, but if they are small, it shouldn't be so terrible to just make them (I can offer to do this). On the other hand, maybe it could work. If I fork to my username, I could work on the site, and then we would merge into the upstream here, and *then* rename to `usrse.github.io`. If that's okay with you, I can do that too :)

**Pull request merged**